icu/icu-59.1-msys2.patch

```diff
@@ -528,18 +37,111 @@ diff -durN icu.orig/source/tools/pkgdata/pkgtypes.h icu/source/tools/pkgdata/pkg
+ #	$(RM) $@ && cp ${<F} $@
+ %.$(SO): %$(SO_TARGET_VERSION_MAJOR).$(SO)
+-	ln -fs $(subst cyg,lib,${<F}).$(A) $(subst cyg,lib,$@).$(A)
++	ln -fs $(subst msys-,lib,${<F}).$(A) $(subst msys-,lib,$@).$(A)
+ 
+ ## Install libraries as executable
+ INSTALL-L=$(INSTALL_PROGRAM)
+--- icu/source/config/mh-cygwin.orig	2021-10-28 18:04:57.000000000 +0200
++++ icu/source/config/mh-cygwin	2022-05-12 09:09:19.521071800 +0200
+@@ -5,6 +5,11 @@
+ ## Copyright (c) 2001-2010, International Business Machines Corporation and
+ ## others. All Rights Reserved.
+ 
++# On Windows we generally have the DLLs in the bin directory rather than the lib directory.
++# This setting moves the ICU DLLs into the bin folder for MinGW/MSYS2 when "make install" is run.
++# If you prefer to have the DLLs in the lib folder, then set this to NO instead.
++MINGW_MOVEDLLSTOBINDIR = YES
++
+ ## Commands to generate dependency files
+ GEN_DEPS.c=	$(CC) -E -MM $(DEFS) $(CPPFLAGS)
+ GEN_DEPS.cc=	$(CXX) -E -MM $(DEFS) $(CPPFLAGS) $(CXXFLAGS)
+@@ -41,8 +46,8 @@
+ ## Commands to make a shared library
+ # $(subst) won't work in shell script.  #M# at the end omits these
+ # lines from the shell script (Make only)
+-SHLIB.c=	$(CC) $(CFLAGS) $(LDFLAGS) -shared $(LD_SOOPTIONS) -Wl,--enable-auto-import -Wl,--out-implib=$(subst cyg,lib,$@).$(A)#M#
+-SHLIB.cc=	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $(LD_SOOPTIONS) -Wl,--enable-auto-import -Wl,--out-implib=$(subst cyg,lib,$@).$(A)#M#
++SHLIB.c=	$(CC) $(CFLAGS) $(LDFLAGS) -shared $(LD_SOOPTIONS) -Wl,--enable-auto-import -Wl,--out-implib=$(subst msys-,lib,$@).$(A)#M#
++SHLIB.cc=	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $(LD_SOOPTIONS) -Wl,--enable-auto-import -Wl,--out-implib=$(subst msys-,lib,$@).$(A)#M#
+ # the #SH# lines don't go through further postprocessing. 
+ # so, they must be valid shell script starting from column 4
+ #SH#SHLIB_c="$(CC) $(CFLAGS) $(LDFLAGS) -shared "
+@@ -60,7 +65,7 @@
+ SO = dll
+ ## Non-shared intermediate object suffix
+ STATIC_O = ao
+-LIBICU = cyg$(ICUPREFIX)
++LIBICU = msys-$(ICUPREFIX)
+ ## To have an import library is better on Cygwin
+ IMPORT_LIB_EXT = .dll.a
+ 
+@@ -94,8 +99,8 @@
+ FINAL_SO_TARGET=$(basename $(SO_TARGET))$(SO_TARGET_VERSION_MAJOR).$(SO)
+ MIDDLE_SO_TARGET=$(FINAL_SO_TARGET)
+ 
+-IMPORT_LIB = $(subst cyg,lib,$(FINAL_SO_TARGET)).$(A)#M#
+-MIDDLE_IMPORT_LIB = $(subst cyg,lib,$(basename $(SO_TARGET)))$(SO_TARGET_VERSION_MAJOR).$(SO).$(A)#M#
++IMPORT_LIB = $(subst msys-,lib,$(FINAL_SO_TARGET)).$(A)#M#
++MIDDLE_IMPORT_LIB = $(subst msys-,lib,$(basename $(SO_TARGET)))$(SO_TARGET_VERSION_MAJOR).$(SO).$(A)#M#
+ FINAL_IMPORT_LIB = $(IMPORT_LIB)#M#
+ 
+ # The following is for Makefile.inc's use.
+@@ -105,7 +110,7 @@
+ #%$(SO_TARGET_VERSION_MAJOR).$(SO): %$(SO_TARGET_VERSION).$(SO)
+ #	$(RM) $@ && cp ${<F} $@
+ %.$(SO): %$(SO_TARGET_VERSION_MAJOR).$(SO)
+-	ln -fs $(subst cyg,lib,${<F}).$(A) $(subst cyg,lib,$@).$(A)
++	ln -fs $(subst cmsys-yg,lib,${<F}).$(A) $(subst msys-,lib,$@).$(A)
```

Comment:
This seems to be broken for 32-bit build
```suggestion
+	ln -fs $(subst msys-,lib,${<F}).$(A) $(subst msys-,lib,$@).$(A)
```

Reply:
Oops, sorry

Reply:
Fixed in master 9b676a2c38862